### ISSUE TITLE
tracing_opentelemetry: Version of `OpenTelemetrySpanExt::set_parent` that returns `self`

### ISSUE THREAD

**Issue by joshtriplett** [tokio-rs/tracing#1412]
### Crates

`tracing_opentelemetry`

### Motivation

I'd like to create a span, set an OpenTelemetry context as the parent, and then use the span in calls like `instrument`, without having to create a named let binding for the span.

### Proposal

I'd like to propose a function `OpenTelemetrySpanExt::with_parent`, which acts exactly like `set_parent`, but accepts `self` and returns `Self`.

```rust
fn with_parent(self, cx: opentelemetry::Context) -> Self {
    self.set_parent(cx);
    self
}
```

Such a function would allow code like this:

```rust
async { ... }.instrument(info_span!("span details").with_parent(context)).await
```

### Alternatives

The tracing macros could have "native" support for OpenTelemetry contexts as parents:

```rust
async { ... }.instrument(info_span!(parent: context, "span details")).await
```

That seems much harder to implement, though.

**Comment by hawkw** [comment]
> ### Alternatives
> 
> The tracing macros could have "native" support for OpenTelemetry contexts as parents:

I think it's unlikely that we'll implement this approach --- the intention is that `tracing` and `tracing-core` should only be a facade layer that's unaware of specific ways of collecting traces or other systems we're integrating with. Therefore, I think adding a new combinator to `tracing-opentelemetry`'s extension trait seems like the right way to do this! :thumbsup:

**Comment by joshtriplett** [comment]
@hawkw I agree entirely, I just mentioned it because it's an alternative. (It'd require some kind of indirection through a trait, I think, so that `tracing` wouldn't have to know about `tracing-opentelemetry`.)

I think `with_parent` seems much simpler.

**Comment by jtescher** [comment]
@joshtriplett yeah `with_parent` sounds good to me!
